### PR TITLE
workspace: enforce crate/directory naming scheme

### PR DESCRIPTION
**Pull request by hinto-janai**
Implements the naming scheme defined in https://github.com/Cuprate/cuprate/pull/144.

- All directories have no prefix/suffix
- All crates names are prefixed with `cuprate-`
- Some crates are renamed

All crate names:
- cuprate-fast-sync
- cuprate-consensus-rules
- cuprate-cryptonight
- cuprate-helper
- cuprate-epee-encoding
- cuprate-fixed-bytes
- cuprate-levin
- cuprate-wire
- cuprate-p2p
- cuprate-p2p-core
- cuprate-dandelion-tower
- cuprate-address-book
- cuprate-blockchain
- cuprate-database
- cuprate-txpool
- cuprate-pruning
- cuprate-test-utils
- cuprate-types
- cuprate-json-rpc
- cuprate-rpc-types
- cuprate-rpc-interface

### CODE REVIEW

**Comment by hinto-janai** [comment]
@Boog900 This is going to cause merge conflicts in pretty much every open PR. Should probably merge close-to-ready ones first so I can deal with the conflicts here.

**Comment by Boog900** [comment]
@hinto-janai The next PR to merge will probably be #132 but it's a pretty big review

File: p2p/p2p-core/src/client/timeout_monitor.rs [hunk 1]
```diff
@@ -5,14 +5,15 @@
...
     time::{interval, MissedTickBehavior},
 };
 use tower::ServiceExt;
 use tracing::instrument;
 
+use cuprate_wire::admin::TimedSyncRequest;
+
 use crate::{
     client::{connection::ConnectionTaskRequest, InternalPeerID},
     constants::{MAX_PEERS_IN_PEER_LIST_MESSAGE, TIMEOUT_INTERVAL},
```
Comment (hinto-janai) [inline p2p/p2p-core/src/client/timeout_monitor.rs:19]: Moved this to follow import ordering.

**Review by Boog900 — APPROVED** [review]
Thank you for working on this, couldn't have been fun.

**Pull request merged**